lib/external/math.dart

```diff
@@ -40,4 +50,128 @@ ASTNode funcRand(Runtime runtime, ASTNode self, List<ASTNode> args) {
   }
 
   throw const UnexpectedTypeException('The rand method only takes [String, double, int, and bool] argument types');
+}
+
+ASTNode funcSin(Runtime runtime, ASTNode self, List<ASTNode> args){
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final radian = args[0];
+  
+  if(radian is IntNode || radian is DoubleNode){
+    final DoubleNode result = DoubleNode()..doubleVal = sin(radian is DoubleNode ? radian.doubleVal : radian.intVal);
+    return result;
+  }
+  
+  throw const UnexpectedTypeException('The sin method only accept either double or int argument types');
```

Comment:
For all these exceptions thrown, specify the type passed using the args type. In this case: `radian.type`

Reply:
I've added to the sentence, **but got ${var.type}**